### DISCUSSION TITLE
Do not hard link .osu file, just hard link audio or some resource files

### DISCUSSION THREAD

**Discussion by cdwcgt** [ppy/osu#27094] [Ideas] [answered]
import from stable will use "hard link" technology, that mean lazer and stable use the same file but point to two path

if I edit the map file in one of these clients, another client will also be affected.
in lazer, beatmap meta and beatmap info also store in realm, I'm not sure if what will be affected

So given that .osu files are usually not large, I think we can just copy but not hard link them.

**Comment by SupDos** [comment, marked as answer]
Editing a hardlinked .osu file shouldn't affect the other .osu file, as per [the wiki](https://osu.ppy.sh/wiki/en/Client/Release_stream/Lazer/File_storage#via-hard-links)

> Updating or editing files causes the files to become separate, and will therefore also not affect the other installation, as long as osu! is used to make the changes.

**Reply by cdwcgt** [reply]
ohh, I got it, stable delete the original file which being hard link, and create a new one
so it will not affect lazer one, thank you let me know
